Makefile.in

```diff
@@ -0,0 +1,38 @@
+GAPPATH = @GAPPATH@
+GAPARCH = @GAPARCH@
+
+GAP=$(GAPPATH)/bin/gap.sh
+
+SOURCES =
+SOURCES += src/avl.c
+SOURCES += src/binaryheap.c
+SOURCES += src/datastructures.c
+SOURCES += src/hashfunctions.c
+SOURCES += src/hashmap.c
+SOURCES += src/pairingheap.c
+SOURCES += src/skiplist.c
+SOURCES += src/uf.c
+
+default: bin/@GAPARCH@/datastructures.so
+
+bin/@GAPARCH@/datastructures.so: $(SOURCES)
+	mkdir -p bin/@GAPARCH@
+	$(GAPPATH)/bin/@GAPARCH@/gac -d -o $@ $?
```

Comment:
This build rule is in invalid. It results in a broken (incomplete) `.so` file when rebuilding, as `$?` only includes the files which changed.

Correct would be to use `$<`. But then every time any file needs to be compiled, all files must be recompiled.

Also, there is no dependency tracking here. So if one compiles using `make`, then edits a `.h` file, and enters `make` again, nothing happens.

I am afraid the super simplistic `orb` build system only works there because `orb` uses a single source file. Packages which use more than one need to implement dependency tracking somehow (e.g. by using `automake` or some other build tool, or by imitating the approach used by GAP's `Makefile.rules`.